core/src/wallet/cosmos/explorers/StargateGaiaCosmosLikeBlockchainExplorer.cpp

```diff
@@ -0,0 +1,1310 @@
+    const std::string &account) const
+{
+    std::vector<FuturePtr<BigInt>> balance_promises({getSpendableBalance(account),
+                                                     getDelegatedBalance(account),
+                                                     getUnbondingBalance(account),
+                                                     getPendingRewardsBalance(account)});
+
+    return async::sequence(getContext(), balance_promises)
+        .flatMap<std::shared_ptr<BigInt>>(getContext(), [](auto &vector_of_balances) {
+            BigInt result;
+            for (const auto &balance : vector_of_balances) {
+                result = result + *balance;
+            }
+            return FuturePtr<BigInt>::successful(std::make_shared<BigInt>(result));
+        });
+}
+/// Get Total Balance
+FuturePtr<BigInt> StargateGaiaCosmosLikeBlockchainExplorer::getTotalBalanceWithoutPendingRewards(
+    const std::string &account) const
+{
+    std::vector<FuturePtr<BigInt>> balance_promises(
+        {getSpendableBalance(account), getDelegatedBalance(account), getUnbondingBalance(account)});
+
+    return async::sequence(getContext(), balance_promises)
+        .flatMap<std::shared_ptr<BigInt>>(getContext(), [](auto &vector_of_balances) {
+            BigInt result;
+            for (const auto &balance : vector_of_balances) {
+                result = result + *balance;
+            }
+            return FuturePtr<BigInt>::successful(std::make_shared<BigInt>(result));
+        });
+}
+/// Get total balance in delegation
+FuturePtr<BigInt> StargateGaiaCosmosLikeBlockchainExplorer::getDelegatedBalance(
+    const std::string &account) const
+{
+    const auto endpoint = fmt::format(cosmos::constants::kGrpcDelegationsEndpoint, _ns, _version, account);
+
+    const bool jsonParseNumbersAsString = true;
+    const auto ignoreFailStatusCode = true;
+
+    return _http->GET(endpoint, ACCEPT_HEADER)
+        .json(jsonParseNumbersAsString, ignoreFailStatusCode)
+        .mapPtr<BigInt>(
+            getContext(),
+            [this, endpoint](const HttpRequest::JsonResult &result) -> std::shared_ptr<BigInt> {
+                // Catching http errors because Cosmos code returns 500 when there is no delegation.
+                if (std::get<0>(result)->getStatusCode() >= 300) {
+                return std::make_shared<BigInt>(BigInt::ZERO);
+                }
+                auto &json = *std::get<1>(result);
+                if (!json.HasMember(kDelegationResponses)) {
+                    throw make_exception(
+                        api::ErrorCode::API_ERROR,
+                        fmt::format("The API response from explorer is missing the \"{}\" key", kDelegationResponses));
+                }
+
+                // Handle null
+                if (!json.GetObject()[kDelegationResponses].IsArray()) {
+                    return std::make_shared<BigInt>(BigInt::ZERO);
+                }
```

Comment:
Consistency: do we want different results when `!HasMember()` and `!IsArray()`?

(applies to a few other places)

Reply:
yeah, because:
-  if there is no `delegation_responses` key at all, it's an issue with the explorer and we can't recover from it.
- if everything is correct, and the user has no current delegation at all, then the explorer response will be `"delegation_responses": null`. In this case that means the account has 0 delegated balance.

That's what the (arguably too small) "Handle null" comment means